JustSaying.IntegrationTests/Fluent/MessagingBusBuilderTests.cs

```diff
@@ -0,0 +1,271 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JustSaying.Fluent;
+using JustSaying.Messaging;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Messaging.Monitoring;
+using JustSaying.Models;
+using JustSaying.TestingFramework;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Shouldly;
+using Xunit.Abstractions;
+
+namespace JustSaying.IntegrationTests
+{
+    public class MessagingBusBuilderTests
+    {
+        public MessagingBusBuilderTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+        }
+
+        private ITestOutputHelper OutputHelper { get; }
+
+        [AwsFact]
+        public async Task Can_Create_Messaging_Bus_Fluently_For_A_Queue()
+        {
+            // Arrange
+            var services = new ServiceCollection()
+                .AddLogging((p) => p.AddXUnit(OutputHelper))
+                .AddJustSaying(
+                    (builder) =>
+                    {
+                        builder.Client((options) => options.WithBasicCredentials("accessKey", "secretKey").WithServiceUri(TestEnvironment.SimulatorUrl))
+                               .Messaging((options) => options.WithRegions("eu-west-1"))
+                               .Publications((options) => options.WithQueue<QueueMessage>())
+                               .Subscriptions((options) => options.ForQueue<QueueMessage>())
+                               .Services((options) => options.WithMessageMonitoring(() => new MyMonitor()));
+                    })
+                .AddJustSayingHandler<QueueMessage, QueueHandler>();
+
+            IServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IMessagePublisher publisher = serviceProvider.GetRequiredService<IMessagePublisher>();
+            IMessagingBus listener = serviceProvider.GetRequiredService<IMessagingBus>();
+
+            using (var source = new CancellationTokenSource(TimeSpan.FromSeconds(20)))
+            {
+                // Act
+                listener.Start(source.Token);
+
+                var message = new QueueMessage();
+
+                await publisher.PublishAsync(message, source.Token);
+
+                // Assert
+                while (!source.IsCancellationRequested && !QueueHandler.MessageIds.Contains(message.Id))
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(1), source.Token);
```

Comment:
I would wait in smaller increments than that, e.g. 200ms. It is probably better for test speed, and not noticeably worse for CPU.